libr/core/cmd_debug.c

```diff
@@ -1576,20 +1575,20 @@ static int cmd_debug_map(RCore *core, const char *input) {
 				char *buf = r_file_slurp (input + 2, &sz);
 				//TODO: use mmap here. we need a portable implementation
 				if (!buf) {
-					eprintf ("Cannot allocate 0x%08"PFMT64x" byte(s)\n", map->size);
+					R_LOG_ERROR ("Cannot allocate 0x%08"PFMT64x" byte(s)", map->size);
 					return false;
 				}
```

Comment:
`r_return_val_if_fail (buf, false);`

Reply:
no. r_return must be used for conditions that cannot change at runtime, for example. a function passing a null pointer or negative length, because those patramters can be controlled by the caller.

this case is a condition that can fail in many different ways:
* the file doesnt exist
* the file is too large
* the heap is over

none of these conditions can be controlled at runtime, so the r_return condition **disappears** when you compile r2 in release mode. which means that in those kind of builds that distros use. the code will just segfault if any of the cases above happen. and thats not what we want